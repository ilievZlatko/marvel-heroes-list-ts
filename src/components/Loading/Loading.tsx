import * as React from 'react';
import loadingSVG from '../../assets/images/loading.svg';

/**
 * The Loading component:
 * Shows a nice spinner, informing user there is a process going on.
 * Needs no props.
 */

const Loading: React.SFC = () => {
    const style = {
		display: 'flex',
		justifyContent: 'center',
		alignItems: 'center',
		height: '150px',
		width: '100vw',
		backgroundColor: 'white'
    };
    
    return (
        <div style={style}>
            <img src={loadingSVG} alt="loading"/>
        </div>
    );
};

export default Loading;
