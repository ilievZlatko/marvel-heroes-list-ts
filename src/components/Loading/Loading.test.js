import React from 'react';
import { shallow, mount } from 'enzyme';

import Loading from './Loading';

describe('Loading component', () => {
    it('renders without crashing', () => {
        const wrap = mount(<Loading />);

        expect(wrap).toMatchSnapshot();
    });

    it('should render spinning svg', () => {
        const wrapper = shallow(<Loading />);

        expect(
            wrapper.containsMatchingElement(<img />)
        ).toBeTruthy();
    });
});