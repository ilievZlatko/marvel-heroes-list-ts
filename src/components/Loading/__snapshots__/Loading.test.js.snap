// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component renders without crashing 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <Loading />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": Object {
      "instance": <div
        style="display: flex; height: 150px; background-color: white;"
>
        <img
                alt="loading"
        />
</div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="loading"
          src={undefined}
/>,
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "display": "flex",
          "height": "150px",
          "justifyContent": "center",
          "width": "100vw",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <img
            alt="loading"
/>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "loading",
            "src": undefined,
          },
          "ref": null,
          "rendered": Array [
            undefined,
          ],
          "type": "img",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": Object {
        "instance": <div
          style="display: flex; height: 150px; background-color: white;"
>
          <img
                    alt="loading"
          />
</div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="loading"
            src={undefined}
/>,
          "style": Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "display": "flex",
            "height": "150px",
            "justifyContent": "center",
            "width": "100vw",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <img
              alt="loading"
/>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "loading",
              "src": undefined,
            },
            "ref": null,
            "rendered": Array [
              undefined,
            ],
            "type": "img",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
